### ISSUE TITLE
Porting Tests within modules

### ISSUE THREAD

**Issue by Graham-72** [backdrop-ops/contrib#91]
@quicksketch - If a Drupal module contains tests, should we be very concerned if these tests fail when we port them to Backdrop? Indeed, is it possible to get them all to work? I find that some of the existing tests in core are failing or not completing on my test installation. Should I be worried about this or is it a known issue?

**Comment by docwilmot** [comment]
I find that when porting modules, tests fail either because of API differences in the tests (say the test file still uses `$entity->language` instead of `->langcode`), or because there are errors in your code. Either way, trying to get tests to pass helps. In the first case you learn a lot about how things work, and in the second its great for debugging. The point of the test file is to programmatically run your module through all probable scenarios (create a user, add a 100 nodes, delete some content, disable this, change that, etc etc) something which may be time-consuming or impossible to do manually. So if your test is all green, you can be reasonably reassured that youve covered most problems/bugs even before your module has faced an end-user.

In short, yes, I'd aim for tests passing. 

Personally, in porting the Example modules, I learned more from failing tests than from anything else so far. And proud to say theyre all green on tests, except for one, which I suspect is haunted, and refuses.

**Comment by quicksketch** [comment]
:+1: @docwilmot 

Absolutely, you should work to get all tests passing in your ported modules, but it can often be a challenge to get them all working. Failing tests indicate changes between Backdrop and D7, and they can reveal accidental API changes in your own module or even in core.

**Comment by Graham-72** [comment]
Thank you. Your guidance is understood and much appreciated.
I have just got my port of Libraries API module to pass all its tests so I'm feeling much more confident about the value of tackling these.
